src/dotnet/updater.py

```diff
@@ -84,30 +84,34 @@
     """{% if image.is_sdk %}# telemetry opt out: https://docs.microsoft.com/en-us/dotnet/core/tools/telemetry#how-to-opt-out
 ENV DOTNET_CLI_TELEMETRY_OPTOUT=1{% endif %}
 
-# The MS GPG keys
-COPY microsoft.asc /tmp
-
 RUN mkdir -p /tmp/
 
 {% for pkg in dotnet_packages -%}
 #!RemoteAssetUrl: {{ pkg.url }}
 COPY {{ pkg.name }} /tmp/
 {% endfor %}
-COPY prod.repo /tmp
 
 # Workaround for https://github.com/openSUSE/obs-build/issues/487
 RUN zypper --non-interactive install --no-recommends sles-release
 
 # Importing MS GPG keys
+COPY microsoft.asc /tmp
 RUN rpm --import /tmp/microsoft.asc
 
 RUN zypper --non-interactive install --no-recommends libicu /tmp/*rpm
 
-RUN cp /tmp/prod.repo /etc/zypp/repos.d/microsoft-dotnet-prod.repo
-
+COPY prod.repo /etc/zypp/repos.d/microsoft-dotnet-prod.repo
 RUN zypper -n addlock dotnet-host
 
 RUN rm -rf /tmp/* && zypper clean && rm -rf /var/log/*
+
+{% if not image.is_sdk %}
+RUN groupadd app && useradd -d /app -g app -G '' -ms /bin/bash app
+USER app
```

Comment:
This will make the images non-extensible, especially on OBS.

Reply:
The groupadd is important to keep aligned with Microsoft official version.
Can we just remove the `USER app` line?

In the ASP.NET 8.0 template generated, they do add this line in the Dockerfile generated (but it's currently failing with BCI because app user is undefined)

Reply:
ah, so I misunderstood, the aspnet container doesn't specify the user by itself. I'll remove this for now.